File: Data Structures & Algorithms/maximum-sum-circular-subarray/submission-0.cpp
class Solution {
   public:
    int maxSubarraySumCircular(vector<int>& nums) {
        int globMax = nums[0], globMin = nums[0];
        int curMax = 0, curMin = 0, total = 0;

        for (int& n : nums) {
            curMax = max(curMax + n, n);
            curMin = min(curMin + n, n);
            total += n;
            globMax = max(globMax, curMax);
            globMin = min(globMin, curMin);
        }

        return globMax > 0 ? max(globMax, total - globMin) : globMax;
    }
};

File: Data Structures & Algorithms/sliding-window-maximum/submission-0.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:     
        l, r = 0, k
        result = []
        while r <= len(nums):
            print(nums[l:r])
            result.append(max(nums[l:r]))
            l += 1
            r += 1

        return result